File: 2021/test/day_15_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_15"

class Day15Test < Minitest::Test
  def test_part1
    day = Day15.new

    assert_equal 40, day.part1(input("day_15_example_1"))
    assert_equal 619, day.part1(input("day_15"))
  end

  def test_part2
    day = Day15.new

    assert_equal 315, day.part2(input("day_15_example_1"))
    assert_equal 2922, day.part2(input("day_15"))
  end
end

File: 2022/test/day_15_test.exs
defmodule Day15Test do
  use ExUnit.Case

  import TestHelper
  import Day15

  test "part1" do
    assert part1(input("day_15_example_1"), 10) == 26
    assert part1(input("day_15"), 2_000_000) == 4_919_281
  end

  test "part2" do
    assert part2(input("day_15_example_1"), 20) == 56_000_011
    assert part2(input("day_15"), 4_000_000) == 12_630_143_363_767
  end
end
